src/engraving/libmscore/textedit.cpp

```diff
@@ -357,21 +365,14 @@ bool TextBase::edit(EditData& ed)
             return true;
 
         case Qt::Key_Left:
-            if (!_cursor->movePosition(ctrlPressed ? TextCursor::MoveOperation::WordLeft : TextCursor::MoveOperation::Left,
-                                       mm) && type() == ElementType::LYRICS) {
-                return false;
-            }
-            s.clear();
-            break;
-
         case Qt::Key_Right:
-            if (!_cursor->movePosition(ctrlPressed ? TextCursor::MoveOperation::NextWord : TextCursor::MoveOperation::Right,
-                                       mm) && type() == ElementType::LYRICS) {
+            if (!toTextBase(ed.element)->moveCursor(_cursor, ed.key, ctrlPressed, mm) && type() == ElementType::LYRICS) {
```

Comment:
moveCursor( ) shouldn't be what's responsible for mapping keys to directions, IMO